### PR TITLE
Add CLR runtime version and framework description

### PR DESCRIPTION
**Pull request by simonbowring**
## Goal

- Add the CLR runtime version to the device section of the payload. This will be added to reports for apps that support .NET Standard 2.0 and above, such as .NET Core 2.0 and above, or any version of the .NET Framework.
- Add the framework description. The framework description is only available for .NET 4.5 onwards and will not return does not return an accurate value for .NET Core [until version 3.0](https://github.com/dotnet/coreclr/issues/22844) so will not be set for .NET Core 1 and 2.

The `runtimeVersions` part of the report payload will look something like this:
```json
runtimeVersions {
  "dotnet": ".NET Framework 4.7.3416.0",
  "dotnetClr": "4.0.30319.42000"
}
``` 
## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: src/Bugsnag/Payload/Device.cs [hunk 1]
```diff
@@ -44,5 +46,23 @@ private static string OsName
...
+
+    private static Dictionary<string, string> RuntimeVersions
+    {
+      get
+      {
+        var runtimeVersions = new Dictionary<string, string>();
+#if NET45
+        // The framework description is only available from .NET 4.5 onwards and will not return
+        // an accurate value for .NET Core until version 3.0 
+        runtimeVersions.AddToPayload("dotnet", RuntimeInformation.FrameworkDescription);
```
Comment (tomlongridge) [inline src/Bugsnag/Payload/Device.cs]: The [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.frameworkdescription) for `RuntimeInformation.FrameworkDescription` says the minimums are:

* .NET Core: 1.0
* .NET Framework: 4.7.1
* .NET Standard: 1.1

Should we not use these preprocessor directives to condition each one?
```
#if NETCOREAPP1_0 || NET471 || NETSTANDARD1_1
```

File: tests/Bugsnag.Tests/Payload/DeviceTests.cs [hunk 2]
```diff
@@ -21,12 +22,21 @@ public void SpecifyHostName()
...
       var device = new Device("hostname");
       Assert.NotNull(device["time"]);
     }
 
+    [Fact]
+    public void IncludesRuntimeVersions()
+    {
+      var device = new Device("hostname");
+      var runtimeVersions = (Dictionary<string, string>) device["runtimeVersions"];
+      Assert.NotNull(runtimeVersions["dotnet"]);
```
Comment (tomlongridge) [inline tests/Bugsnag.Tests/Payload/DeviceTests.cs]: Can you add a regex test to ensure it contains `\d+\.` 4 times? This is what was tested in other notifiers.

File: src/Bugsnag/Payload/Device.cs (hunk 1 above)
Comment (simonbowring) [reply src/Bugsnag/Payload/Device.cs]: I think we should only include it for the .NET Framework for now because this won't show an accurate version for the .NET Core/.NET Standard [until version 3.0 is released](https://github.com/dotnet/coreclr/issues/22844). 

As for .NET 4.5 vs 4.7.1, the code compiles ok for 4.5 so it might be worth seeing what is actually recorded when running with that version.

**Comment by simonbowring** [comment]
Very old PR!
